protocols/kad/src/handler.rs

```diff
@@ -611,19 +605,13 @@ where
     type InEvent = KademliaHandlerIn<TUserData>;
     type OutEvent = KademliaHandlerEvent<TUserData>;
     type Error = io::Error; // TODO: better error type?
-    type InboundProtocol = Either<KademliaProtocolConfig, upgrade::DeniedUpgrade>;
```

Comment:
That is not quite correct. You still need to set the associated type but just to `KademliaProtocolConfig` instead of the composed `Either`.